src/atoms/weavemol.jl

```diff
@@ -1,13 +1,16 @@
 # some commentary
 
 # docstring...
+
 mutable struct WeaveMol <: AbstractAtoms
     smiles::String
     elements::Vector{String}
-    atom_features::Vector{SomethingOrOther} # I need to look more carefully to figure this out, heh
-    pair_features::Vector{SomethingOrOther}
+    atom_features::Vector{SomethingOrOther} # Details TBD
+    # pairs encapsulates the PairFeature features for a given pair of atoms in a molecule represented using PairIndexes
+    pairs::Vector{PairFeature}
```

Comment:
I don't think this type is correct; it should be the type associated with encoded pair features, not the descriptor objects (I know you said you had locally retyped some things though so maybe that's what's going on?)

Reply:
Oh right, yeah. I think I missed out including that bit in the last commit, sorry. Fwiw, I've currently (locally) defined it to be a `Union{Matrix{<:Real}, Nothing}`.
That works, right?
(While the 3D matrix idea has an intuitive appeal to it, it still feels like it could be lot less memory efficient.)

Reply:
if we're not going the 3D matrix route (personally, I think it could be okay as the more "user-facing" thing and then we'd just need some kind of "interfacing" function to convert it into whatever needs to be fed into the model, but that's not so bad), then it needs to be something that can accommodate any "shape" of encoded pair features. Matrix works if they encode to scalars, but if they're vectors, or even (at some future date) matrices themselves, then this won't work unless it's something like a matrix of vectors, at which point we should probably go the higher-D matrix route anyway...

Reply:
On that front, I agree. If/when we think of a better representation, we can probably update it, I guess. I'll update it to a 3D matrix then.

Reply:
Resolved in b456702.